src/operator/operator_util.cc

```diff
@@ -0,0 +1,831 @@
+/*!
+ *  Copyright (c) 2015 by Contributors
+ * \file operator_util.cc
+ *  Implementation of operator util.
+ */
+#include <mxnet/operator_util.h>
+#include <mxnet/operator.h>
+#include <mxnet/ndarray.h>
+#include <mxnet/engine.h>
+#include <vector>
+#include <mutex>
+#include "./operator_common.h"
+
+namespace mxnet {
+namespace op {
+
+class SimpleOpPropBase;
+class SimpleUnaryOpProp;
+class SimpleBinaryOpProp;
+
+class SimpleOpRegEntryImpl : public SimpleOpRegEntry {
+ public:
+  TSelf& set_enable_scalar(
+      bool enable_scalar,
+      SimpleOpScalarOption type_mask) override {
+    std::lock_guard<std::mutex> lock(mutex_);
+    enable_scalar_ = enable_scalar;
+    scalar_type_mask_ = type_mask;
+    CHECK(!enable_kwargs_ || !enable_scalar_)
+        << "Cannot have both kwargs and scalar arguments";
+    return *this;
+  }
+
+  TSelf& set_enable_kwargs(bool enable_kwargs) override {
+    std::lock_guard<std::mutex> lock(mutex_);
+    enable_kwargs_ = enable_kwargs;
+    CHECK(!enable_kwargs_ || !enable_scalar_)
+        << "Cannot have both kwargs and scalar arguments";
```

Comment:
This simplifies things a bit If you switch to kwargs, use kwargs for all the arguments, including possible scalar. Scalar is used for supporting some quick pass in of real arguments